R/SubsetDefinitions.R

```diff
@@ -329,7 +329,18 @@ addCohortSubsetDefinition <- function(cohortDefinitionSet,
   subsetDefinitionCopy <- cohortSubsetDefintion$clone(deep = TRUE)
 
   # Remove any cohorts that use this id
-  if (!is.null(existingSubsetDefinitions[subsetDefinitionCopy$definitionId][[1]])) {
+  findSubsetIndexById <- function(existingSubsetDefinitions, id) {
+    ind <- sapply(existingSubsetDefinitions, function(y) existingSubsetDefinitions$definitionId == id)
+    if (length(ind) > 0) {
```

Comment:
A test is needed for adding multiple subset definitions as I'm not sure how this code will work. I think you mean `y$definitionId == id` in the anonymous function above?

This fails when none are found because ind will be a list containing the element`logical(0)`.

I think a for loop also makes more sense because we enforce the rule that `definitionId` is unique, so this search should terminate upon finding the index.

```
findSubsetIndexById <- function(existingSubsetDefinitions, id) {
    for (i in 1:length(existingSubsetDefinitions)) {
        if (existingSubsetDefinitions[[i]]$definitionId == id)
            return(i)
    }
    return(NA)
  } 
```